src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,428 @@
+                updateCohort(_.set(`criteriaGroups.${index}.criteria.${criteriaGroup.criteria.length}`, criteria));
+              }
+            },
+          }),
+        ]),
+      ]),
+      div(
+        {
+          style: {
+            paddingTop: 5,
+            marginTop: 10,
+            marginBottom: 10,
+            borderTop: `1px solid ${colors.dark(0.35)}`,
+            display: 'flex',
+            justifyContent: 'flex-end',
+            fontWeight: 'bold',
+          },
+        },
+        [div({ style: { marginRight: 10 } }, [`Group count: ${criteriaGroup.count}`])]
+      ),
+    ]
+  );
+};
+
+type CohortGroupsProps = {
+  cohort: Cohort | undefined;
+  datasetDetails: DatasetResponse;
+  updateCohort: CohortUpdater;
+};
+const CohortGroups = ({ datasetDetails, cohort, updateCohort }: CohortGroupsProps) => {
+  return div({ style: { width: '47rem' } }, [
+    cohort == null
+      ? 'No cohort found'
+      : div([
+          _.map(
+            ([index, criteriaGroup]) =>
+              h(Fragment, { key: criteriaGroup.name }, [
+                h(CriteriaGroupView, { index, criteriaGroup, updateCohort, cohort, datasetDetails }),
+                div({ style: { marginTop: '1rem', display: 'flex', alignItems: 'center' } }, [
+                  div(
+                    {
+                      style: {
+                        height: 45,
+                        width: 45,
+                        backgroundColor: 'white',
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                        borderRadius: '50%',
+                        border: `1px solid ${colors.dark(0.35)}`,
+                        fontStyle: 'italic',
+                      },
+                    },
+                    ['And']
+                  ),
+                  div({
+                    style: { marginLeft: 5, flexGrow: 1, borderTop: `1px solid ${colors.dark(0.35)}` },
+                  }),
+                ]),
+              ]),
+            Utils.toIndexPairs(cohort.criteriaGroups)
+          ),
+        ]),
+  ]);
+};
+
+const editorBackgroundColor = colors.light(0.7);
+
+type CohortEditorContentsProps = {
+  updateCohort: CohortUpdater;
+  cohort: Cohort;
+  datasetDetails: DatasetResponse;
+  onStateChange: OnStateChangeType;
+};
+const CohortEditorContents = ({ updateCohort, cohort, datasetDetails, onStateChange }: CohortEditorContentsProps) => {
+  return div(
+    {
+      style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` },
+    },
+    [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: () => {
+              onStateChange(new HomepageState());
+            },
+            'aria-label': 'cancel',
+          },
+          [icon('circle-chevron-left', { size: 32, className: 'regular', style: { marginRight: 5 } })]
```

Comment:
Opt nit: Why marginRight here rather than marginLeft on the following component? (I tend to avoid `marginRight` and `marginBottom` because they tend to imply things about the following elements as opposed to this element.)